### PR TITLE
feat(select): gh-select

### PR DESCRIPTION
**Pull request by tomheller**
Select output from the workshop day. Let's improve upon this

### CODE REVIEW

**Comment by thomaspink** [comment]
Issue: #16

**Comment by thomaspink** [comment]
Blocked until styles have been pushed to css-groundhog

File: src/lib/select/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Select
+
+`<gh-select>` is like the native `<select>` a form control for selecting a value from a list of options. It is also desinged to work with Angular Forms.
```
Comment (ffriedl89) [inline src/lib/select/README.md]: typo

File: src/lib/select/select.scss [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@import '@dynatrace/groundhog/src/common/mixins';
+@import '@dynatrace/groundhog/src/select/select-common';
+
+@include cdk-overlay();
+
+$gh-select-panel-max-height: 242px !default;
+$gh-select-panel-min-width: 112px !default;
+$gh-select-panel-max-width: 280px !default;
+
+// TODO @ffriedl89: move variables (let's first discuss how to handle them)
```
Comment (ffriedl89) [inline src/lib/select/select.scss]: remove this

File: src/lib/select/select.scss [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  backface-visibility: hidden;
+
+  transform-origin: left top;
+
+  min-width: $gh-select-panel-min-width;
+  max-width: $gh-select-panel-max-width;
+  overflow: auto;
+  -webkit-overflow-scrolling: touch;   // for momentum scroll on mobile
+
+  max-height: $gh-select-panel-max-height;
```
Comment (ffriedl89) [inline src/lib/select/select.scss]: move this to component - to have magic number 242 only once

File: src/lib/select/select.ts [hunk 4]
```diff
@@ -0,0 +1,830 @@
...
+  /** Emits whenever the component is destroyed. */
+  private _destroy = new Subject();
+
+  /** Comparison function to specify which option is displayed. Defaults to object equality. */
+  private _compareWith = (o1: any, o2: any) => o1 === o2;
+
+  /** The scroll position of the overlay panel, calculated to center the selected option. */
+  private _scrollTop = 0;
+
+  private _connectionPair: ConnectionPositionPair;
```
Comment (ffriedl89) [inline src/lib/select/select.ts]: comment

File: src/lib/select/select.ts [hunk 5]
```diff
@@ -0,0 +1,830 @@
...
+
+  /** Deals with the selection logic. */
+  _selectionModel: SelectionModel<GhOption>;
+
+  /** The IDs of child options to be passed to the aria-owns attribute. */
+  _optionIds: string = '';
+
+  /** The aria-describedby attribute on the select for improved a11y. */
+  _ariaDescribedby: string; // TODO @thomaspink: Implement when adding support for angular forms
```
Comment (ffriedl89) [inline src/lib/select/select.ts]: remove todo

File: src/lib/select/select.ts [hunk 6]
```diff
@@ -0,0 +1,830 @@
...
+
+  /** Whether the panel's animation is done. */
+  _panelDoneAnimating: boolean = false;
+
+  /**
+   * The y-offset of the overlay panel in relation to the trigger's top start corner.
+   * This must be adjusted to either -1 (if overlay opend below the trigger)
+   * or 1 (if overlay opens on top) to overlap the border of the overlay and trigger.
+   */
+  _offsetY = -1;
```
Comment (ffriedl89) [inline src/lib/select/select.ts]: remove

**Comment by thomaspink** [comment]
@ffriedl89 applied feedback. Pls review

**Pull request merged**